scst/src/dev_handlers/scst_disk.c

```diff
@@ -374,6 +374,17 @@ static enum scst_exec_res disk_exec(struct scst_cmd *cmd)
 		}
 	}
 
+	/*
+	 * If we are passing thru a INQUIRY VPD=0x83 (device identification) then
+	 * we will call scst_replace_port_info on success in scst_pass_through_cmd_done.
+	 * This will run in interrupt context, so we should not perform operations that
+	 * involve mutexes.  Call scst_lookup_tg_id here intead and save the output.
+	 */
+	if (unlikely((cmd->cdb[0] == INQUIRY) &&
```

Comment:
I think it would be nice if you made the condition into a separate function and put it somewhere in scst.h, for example:

```
static inline bool scst_cmd_inquired_dev_info(struct scst_cmd *cmd)
{
        return (cmd->cdb[0] == INQUIRY) && ((cmd->cdb[1] & 0x01/*EVPD*/) == 0x01) &&
	          (cmd->cdb[2] == 0x83/*device identification*/)))
}
```
So you can use this helper here, and in scst_replace_port_info() as well.

And I think you missed setting the `tg_id` in scst_do_real_exec() before scst_scsi_exec_async():

```
	if (unlikely(scsi_dev == NULL)) {
		PRINT_ERROR("Command for virtual device must be "
			"processed by device handler (LUN %lld)!",
			(unsigned long long)cmd->lun);
		goto out_error;
	}

	TRACE_DBG("Sending cmd %p to SCSI mid-level dev %d:%d:%d:%lld", cmd,
		  scsi_dev->host->host_no, scsi_dev->channel, scsi_dev->id,
		  (u64)scsi_dev->lun);
        
+       if (unlikely(scst_cmd_inquired_dev_info(cmd))
+               cmd->tg_id = scst_lookup_tg_id(dev, cmd->tgt);
	
	rc = scst_scsi_exec_async(cmd, cmd, scst_pass_through_cmd_done);
```

Reply:
Thanks for the review!  Have now updated PR to include the requested changes (albeit with a slight rename).

Reply:
Hi Brian,

Thank you for the patch!

Thanks,
Gleb